pkgs/create-neon/data/versions.json

```diff
@@ -1,4 +1,25 @@
 {
   "neon": "1",
-  "cargo-cp-artifact": "0.1"
+  "neonCLI": "0.1.64",
+  "neonLoad": "0.1.64",
+  "typescript": "5.3.3",
+  "typesNode": "20.11.16",
+  "tsconfigNode": {
+    "major": "20",
+    "semver": "20.1.4",
+    "module": "16"
+  },
+  "node": "20",
+  "actions": {
+    "checkout": "v3",
+    "githubScript": "v7",
+    "setupNode": "v3",
+    "setupRust": "v1",
+    "cargoInstall": "v2",
+    "neonBuild": "v0.9",
+    "neonPublish": "v0.4.1",
+    "dotenv": "v1",
+    "ghRelease": "v1",
+    "releaseDownloader": "v1.7"
```

Comment:
Does GitHub offer any protections on these tags changing? They _didn't_ used to, but I'm not sure if they added it.

If not, can we use explicit SHA on unofficial actions?

Reply:
I divided the `versions.json` metadata into two categories, one for verified and one for unverified actions:

```json
  "actions": {
    "verified": {
      "checkout": "v3",
      "githubScript": "v7",
      "setupNode": "v3",
      "setupRust": "v1",
      "cargoInstall": "v2",
      "neonBuild": "v0.9",
      "neonPublish": "v0.4.1"
    },
    "unverified": {
      "dotenv": {
        "tag": "v1.1.2",
        "sha": "d1cd55661714e830a6e26f608f81d36e23424fed"
      },
      "ghRelease": {
        "tag": "v2.0.4",
        "sha": "9d7c94cfd0a1f3ed45544c887983e9fa900f0564"
      },
      "releaseDownloader": {
        "tag": "v1.10",
        "sha": "c39a3b234af58f0cf85888573d361fb6fa281534"
      }
    }
  }
```

And I changed the template to generate an explicit SHA reference with a comment noting the version tag, e.g.:

```
- name: Set Environment Variables
  uses: falti/dotenv-action@d1cd55661714e830a6e26f608f81d36e23424fed # v1.1.2
```